### PR TITLE
[WIP][ADD] Add wocg command.

### PR DESCRIPTION
**Pull request by adrienpeiffer**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        'Programming Language :: Python :: 2.7',
+        'Framework :: Django :: 1.11',
+    ],
+    setup_requires=[
+        'setuptools-git',
+        'setuptools-scm',
+    ],
+    entry_points='''
+        [console_scripts]
+        wocg=sync_git_weblate.main:main
```
Comment (sbidoul) [inline setup.py]: sync_git_weblate does not look correct

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License GPL-3.0 or later (http://www.gnu.org/licenses/gpl.html).
+
+from setuptools import setup, find_packages
+
+
+setup(
+    name='weblate-odoo-component-generator',
+    description='weblate-odoo-component-generator',
```
Comment (sbidoul) [inline setup.py]: TODO (or remove description field?)

File: wocg/main.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            logger.info('WOCG: Main component not found for project %s'% project.name)
+            continue
+        logger.info('WOCG: Main component found for project %s : %s' % (
+            project.name, main_subproject.name))
+        repo = 'weblate://%s/%s' % (project.slug, main_subproject.slug)
+        main_subproject_addon_name = _get_addon_name(main_subproject)
+        repo_path = os.path.join(svn_dir, project.slug, main_subproject.slug)
+        os.chdir(repo_path)
+        existing_sub_projects_slug = _get_all_subprojects_slug(project)
+        addons = get_translatable_addons()
```
Comment (sbidoul) [inline wocg/main.py]: may be a bit more robust is starting from the filemask of the main component which should match `r"^(?P<addons_dir>.*)/(?P<addon_name>.*?)/i18n/\*\.po$"` and then listing `addons_dir` to find addons to translate (no need to look at manifest etc).

**Review by sbidoul — COMMENTED** [review]
joli

File: wocg/main.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+def _get_slug_name(project, addon):
+    return "%s-%s" % (project.slug, addon)
+
+
+def _get_addon_name(component):
+    return component.slug.split('-')[-1]
+
+
+def main():
+    all_projects = Project.objects.prefetch_related('source_language')
```
Comment (sbidoul) [inline wocg/main.py]: Should we not pass the project name as argument to the script? There might be non odoo projects in the django instance.

File: wocg/manifest.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+
+import os
+
+MANIFEST_NAMES = ('__openerp__.py', '__manifest__.py', '__terp__.py')
+
+
+class NoManifestFound(Exception):
+    pass
```
Comment (sbidoul) [inline wocg/manifest.py]: this is not used here

File: wocg/manifest.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    addons_dirs = []
+    candidate_addons_dirs = (
+        os.path.join('odoo', 'addons'),
+        'odoo_addons',
+        '.',
+    )
+    for addons_dir in candidate_addons_dirs:
+        if os.path.isdir(addons_dir):
+            addons_dirs.append(addons_dir)
+    return addons_dirs
```
Comment (sbidoul) [inline wocg/manifest.py]: this is not necessary here

File: wocg/main.py [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        sub_projects_slug.append(sub_project.slug)
+    return sub_projects_slug
+
+
+def _get_slug_name(project, addon):
+    return "%s-%s" % (project.slug, addon)
+
+
+def _get_addon_name(component):
+    return component.slug.split('-')[-1]
```
Comment (sbidoul) [inline wocg/main.py]: not used?

**Review by sbidoul — COMMENTED** [review]
This looks almost finished, do you have other things to do in mind for this script?

File: wocg/main.py (hunk 4 above)
Comment (adrienpeiffer) [reply wocg/main.py]: We can. We also need to check if it's possible to launch wocg from post update script from the main component

File: wocg/manifest.py (hunk 6 above)
Comment (adrienpeiffer) [reply wocg/manifest.py]: Indeed

**Comment by adrienpeiffer** [comment]
@sbidoul Did I adapt this one to work with the version 11.0 ?

**Comment by sbidoul** [comment]
@adrienpeiffer 

> Did I adapt this one to work with the version 11.0 ?

I don't understand the question

**Comment by adrienpeiffer** [comment]
Bad PR sorry. I mean about acsoo makepot

**Comment by adrienpeiffer** [comment]
@benwillig :gift:

**Pull request merged**